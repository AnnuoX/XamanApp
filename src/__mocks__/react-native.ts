/* eslint-disable */

const ReactNative = require('react-native');
const crypto = require('crypto');

ReactNative.NativeModules.CryptoModule = {
    randomKeySync: jest.fn((len: number) => crypto.randomBytes(len).toString('hex').toUpperCase()),
    randomKey: jest.fn((len: number) => Promise.resolve(crypto.randomBytes(len).toString('hex').toUpperCase())),
    sha512: jest.fn((value) => Promise.resolve(crypto.createHash('sha512').update(value).digest('hex'))),
    sha256: jest.fn((value) => Promise.resolve(crypto.createHash('sha256').update(value).digest('hex'))),
    sha1: jest.fn((value) => Promise.resolve(crypto.createHash('sha1').update(value).digest('hex'))),
    hmac256: jest.fn((value, key) =>
        Promise.resolve(crypto.createHmac('sha256', Buffer.from(key, 'hex')).update(value).digest('hex')),
    ),
    encrypt: jest.fn((entry, key) => {
        const ivHex = crypto.randomBytes(16).toString('hex').toUpperCase();
        const keyEnc = crypto.createHash('sha256').update(key).digest();

        const iv = Buffer.from(ivHex, 'hex');
        const cipher = crypto.createCipheriv('aes-256-cbc', keyEnc, iv);
        entry = Buffer.from(entry);
        let crypted = cipher.update(entry, 'utf-8', 'base64');
        crypted += cipher.final('base64');
        return Promise.resolve({ cipher: crypted, iv: ivHex });
    }),
    decrypt: jest.fn((entry, key, iv) => {
        const keyEnc = crypto.createHash('sha256').update(key).digest();
        iv = Buffer.from(iv, 'hex');
        const decipher = crypto.createDecipheriv('aes-256-cbc', keyEnc, iv);
        let decrypted = decipher.update(entry, 'base64');
        decrypted += decipher.final();
        return Promise.resolve(decrypted);
    }),
};

ReactNative.NativeModules.UtilsModule = {
    flagSecure: jest.fn((enable: boolean) => true),
    isRooted: jest.fn(() => false),
    isJailBroken: jest.fn(() => false),
    getTimeZone: jest.fn(() => 'Europe/Amsterdam'),
    hapticFeedback: jest.fn((type: any) => true),
    restartBundle: jest.fn(() => true),
    timeoutEvent: jest.fn((event: string, timeout: number) => true),
    getElapsedRealtime: jest.fn(() => Promise.resolve(123)),
    exitApp: jest.fn(() => true),
};

ReactNative.NativeModules.LocalNotificationModule = {
    setBadge: jest.fn((badge: number) => Promise.resolve()),
    getBadge: jest.fn(() => Promise.resolve(1)),
    complete: jest.fn((messageId: string, show: boolean) => true),
};

ReactNative.NativeModules.AppUpdateModule = {
    checkUpdate: jest.fn(() => Promise.resolve(123)),
    startUpdate: jest.fn(() => Promise.resolve()),
};

ReactNative.NativeModules.SharedPreferencesModule = {
    get: jest.fn((key) => Promise.resolve('value')),
    set: jest.fn((key, value) => Promise.resolve(true)),
    del: jest.fn((key) => Promise.resolve(true)),
};

ReactNative.NativeModules.Toast = {
    showWithGravity: jest.fn((message: string, duration: any, gravity: number) => true),
};

ReactNative.NativeModules.DeviceUtilsModule = {
    layoutInsets: { top: 0, bottom: 0 },
    brand: 'Apple',
    model: 'iPhone13,4',
    osVersion: '15,5',
    getElapsedRealtime: jest.fn(() => Promise.resolve(0)),
    isJailBroken: jest.fn(() => Promise.resolve(false)),
    isRooted: jest.fn(() => Promise.resolve(false)),
    getTimeZone: jest.fn(() => Promise.resolve('Europe/Amsterdam')),
    getLocalSetting: jest.fn(() =>
        Promise.resolve({ delimiter: ',', languageCode: 'en', locale: 'en_US', separator: '.' }),
    ),
};

ReactNative.NativeModules.AppUtilsModule = {
    appVersion: '0.0.1',
    buildNumber: '1',
    isDebug: false,
    isFlagSecure: jest.fn(() => Promise.resolve(false)),
    setFlagSecure: jest.fn((value) => {}),
    restartBundle: jest.fn(() => {}),
    exitApp: jest.fn(() => {}),
    timeoutEvent: jest.fn((key, delay) => {}),
};

ReactNative.NativeModules.UniqueIdProviderModule = {
    getDeviceUniqueId: jest.fn(() => 'e988b7a9-f685-4674-87bc-0ad52a52faa5'),
};

ReactNative.NativeModules.HapticFeedbackModule = {
    trigger: jest.fn((type) => {}),
};

module.exports = ReactNative;
