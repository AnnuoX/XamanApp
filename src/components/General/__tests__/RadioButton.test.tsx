/**
 * @format
 */

import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { TouchableOpacity } from 'react-native';

import { RadioButton } from '..';

describe('[RadioButton]', () => {
    it('renders correctly', () => {
        const onPress = jest.fn();

        const tree = renderer
            .create(
                <RadioButton
                    onPress={onPress}
                    label="label"
                    labelSmall="label small"
                    description="description"
                    checked={false}
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders checked correctly', () => {
        const onPress = jest.fn();

        const tree = renderer
            .create(
                <RadioButton
                    onPress={onPress}
                    label="label"
                    labelSmall="label small"
                    description="description"
                    checked
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls the given function when the radio button is pressed', () => {
        const onPress = jest.fn();
        const tree = renderer.create(<RadioButton label="label" checked onPress={onPress} />);

        const touchInst = tree.root.findByType(TouchableOpacity);

        touchInst.props.onPress();

        expect(onPress).toHaveBeenCalled();
    });
});
