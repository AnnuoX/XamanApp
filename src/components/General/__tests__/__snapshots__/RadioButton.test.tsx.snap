// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RadioButton] renders checked correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F3F6FA",
      "borderColor": "#0030CF",
      "borderRadius": 20,
      "borderWidth": 3,
      "color": "#ACB1C1",
      "flexDirection": "row",
      "justifyContent": "center",
      "left": 0,
      "marginBottom": 20,
      "opacity": 1,
      "padding": 20,
      "right": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#606885",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 26,
            "justifyContent": "center",
            "marginTop": 4,
            "width": 26,
          },
          {
            "borderColor": "#0030CF",
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#0030CF",
            "borderRadius": 8,
            "color": "#0030CF",
            "height": 15,
            "width": 15,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "flex": 6,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 34.285714285714285,
          },
          {
            "fontFamily": "ProximaNova-Extrabld",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      label
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      label small
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 30,
          },
          {
            "marginTop": 4,
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      description
    </Text>
  </View>
</View>
`;

exports[`[RadioButton] renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(96,104,133,0.06)",
      "borderColor": "rgba(96,104,133,0.06)",
      "borderRadius": 20,
      "borderWidth": 3,
      "color": "#ACB1C1",
      "flexDirection": "row",
      "justifyContent": "center",
      "left": 0,
      "marginBottom": 20,
      "opacity": 1,
      "padding": 20,
      "right": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#606885",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 26,
            "justifyContent": "center",
            "marginTop": 4,
            "width": 26,
          },
          false,
        ]
      }
    />
  </View>
  <View
    style={
      {
        "flex": 6,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 34.285714285714285,
          },
          {
            "fontFamily": "ProximaNova-Extrabld",
          },
          {
            "color": "#606885",
          },
        ]
      }
    >
      label
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "ProximaNova-Bold",
            "fontSize": 30,
          },
          {
            "color": "#606885",
          },
        ]
      }
    >
      label small
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 30,
          },
          {
            "marginTop": 4,
          },
          {
            "color": "#606885",
          },
        ]
      }
    >
      description
    </Text>
  </View>
</View>
`;
